src/ui/public/saved_objects/saved_objects_client.js

```diff
@@ -0,0 +1,95 @@
+import { resolve as resolveUrl, format as formatUrl } from 'url';
+import { pick, partial } from 'lodash';
+
+import { SavedObject } from './saved_object';
+
+const join = (...uriComponents) => (
+  uriComponents.filter(Boolean).map(encodeURIComponent).join('/')
+);
+
+export class SavedObjectsClient {
+  constructor($http, basePath) {
+    this._$http = $http;
+    this._apiBaseUrl = `${basePath}/api/saved_objects/`;
+  }
+
+  get(type, id) {
+    if (!type || !id) {
+      return Promise.reject(new Error('requires type and id'));
```

Comment:
Unfortunately, it seems that number 1 is still true, we need to use angular's promise constructor here, which is the default for `$http`, but will need to be passed in via the constructor if we are going to be doing validation here.

See https://digest-test-vhmbhnyquk.now.sh/